[IMP] marketing: sms essentials automated rule v17 (#8957)
odoo/documentation: content/applications/marketing/sms_marketing/essentials/sms_essentials.rst

   :align: center
   :alt: Reporting page in SMS Marketing.

.. tip::
   SMS messages can be sent using automation rules in Odoo. Odoo *Studio* is required to use
   automation rules.

   To install Odoo *Studio*, go to the :menuselection:`Apps application`. Then, using the
   :guilabel:`Search...` bar, search for `studio`.

   If it is not already installed, click :guilabel:`Install`.

   Adding the *Studio* application upgrades the subscription status to *Custom*, which increases the
   cost. Consult `support <https://www.odoo.com/contactus>`_, or reach out to the database's
   customer success manager, with any questions on making the change.

   To use automation rules, navigate in :ref:`developer mode <developer-mode>`, to
   :menuselection:`Settings app --> Technical menu --> Automation section --> Automation Rules`.
   Then, click :guilabel:`New` to create a new rule.

   Enter a name for the automation rule, and select a :guilabel:`Model` to implement this rule on.

   Based on the selection for the :guilabel:`Trigger`, additional fields will populate below. Set
   the :guilabel:`Trigger` to one of the following options:

>    :guilabel:`Values Updated`
>     - :guilabel:`User is set`
>     - :guilabel:`State is set to`
>     - :guilabel:`On archived`
>     - :guilabel:`On unarchived`
> 
>    :guilabel:`Timing Conditions`
>     - :guilabel:`Based on date field`
>     - :guilabel:`After creation`
>     - :guilabel:`After last update`
> 
>    :guilabel:`Custom`
>     - :guilabel:`On save`
>     - :guilabel:`On deletion`
>     - :guilabel:`On UI change`
> 
>    :guilabel:`External`
>     - :guilabel:`On webhook`
> 
>    Other options may appear based on the :guilabel:`Model` selected. For example if the
>    :guilabel:`Calendar Event` model is selected, then the following options appear in addition to
>    those above:
> 
>    :guilabel:`Email Events`
>     - :guilabel:`On incoming message`
>     - :guilabel:`On outgoing message`

Reviewer: Add an empty line above list items and remove 1 space indent :) 
```suggestion
   :guilabel:`Values Updated`

   - :guilabel:`User is set`
   - :guilabel:`State is set to`
   - :guilabel:`On archived`
   - :guilabel:`On unarchived`

   :guilabel:`Timing Conditions`

   - :guilabel:`Based on date field`
   - :guilabel:`After creation`
   - :guilabel:`After last update`

   :guilabel:`Custom`

   - :guilabel:`On save`
   - :guilabel:`On deletion`
   - :guilabel:`On UI change`

   :guilabel:`External`

   - :guilabel:`On webhook`

   Other options may appear based on the :guilabel:`Model` selected. For example if the
   :guilabel:`Calendar Event` model is selected, then the following options appear in addition to
   those above:

   :guilabel:`Email Events`

   - :guilabel:`On incoming message`
   - :guilabel:`On outgoing message`
```

   :align: center
   :alt: Reporting page in SMS Marketing.

.. tip::

>    SMS messages can be sent using automated rules in Odoo. Odoo *Studio* is required to use
>    automated rules. To install it, go to :menuselection:`Apps --> Search...` bar and search for
>    `studio`. If it is not already installed, click :guilabel:`Install`. Adding the *Studio*
>    application upgrades the subscription status to *Custom*, which increases the cost. Consult
>    `support <https://www.odoo.com/contactus>`_ or reach out to the database's customers success
>    manager with any questions on upgrading.

Reviewer 2: ```suggestion
   SMS messages can be sent using automated actions in Odoo. Odoo *Studio* is required to use
   automated actions.
   
   To install Odoo *Studio*, go to :menuselection:`Apps application`. Then, using the
   :guilabel:`Search...` bar, and search for `studio`.
   
   If it is not already installed, click :guilabel:`Install`.
   
   Adding the *Studio* application upgrades the subscription status to *Custom*, which increases the
   cost. Consult `support <https://www.odoo.com/contactus>`_, or reach out to the database's
   customer success manager, with any questions on upgrading.
```

>    To use automated rules, navigate to :menuselection:`Settings app --> Technical menu -->

Reviewer 2: don't forget to mention that users must be in developer mode in order to reveal the Technical Menu...

Author: @ksc-odoo 

This isn't true for Odoo 17.
![2024-04-24_14-12](https://github.com/odoo/documentation/assets/91896312/7aa9ae86-28ac-4053-9d72-ab8c1bb34700)

Reviewer 2: ![Screenshot 2024-04-24 111937](https://github.com/odoo/documentation/assets/73958186/aeb4fb75-7b8d-4e42-bf44-046de4a7f6da)

Author: That's wild! 

@ksc-odoo 

So I replicated both ways (with and without dev mode turned on) for a trial db - (non edu) in 17.0 

I think we go with activating developer mode (to cover all bases)

Thanks for troubleshooting this with me! :+1: 

Tim

>    Automation section --> Automated Rules`. Then, click :guilabel:`New` to create a new rule.

Reviewer 2: ```suggestion
   Automation section --> Automation Rules`. Then, click :guilabel:`New` to create a new rule.
```

>    Enter a name for the automated rule and select a :guilabel:`Model` to implement this rule on.

Reviewer 2: ```suggestion
   Enter a name for the automated rule, and select a :guilabel:`Model` to implement this rule on.
```

>    Once setting the :guilabel:`Model` other options will populate.

Reviewer 2: ```suggestion
   Once setting the :guilabel:`Model` other options populate.
```

>    Set the :guilabel:`Trigger` to one of the following options:

Reviewer 2: might be helpful to mention that some options may appear/not appear, depending on the Model selected

>    :guilabel:`Value Updated`

Reviewer 2: ```suggestion
   :guilabel:`Values Updated`
```

    - :guilabel:`User is set`
    - :guilabel:`State is set to`
    - :guilabel:`On archived`
    - :guilabel:`On unarchived`

> 

Reviewer 2: Is there a reason the 'Email Events' section/options were overlooked here?

   :guilabel:`Timing Conditions`
    - :guilabel:`Based on date field`
    - :guilabel:`After creation`
    - :guilabel:`After last update`

   :guilabel:`Custom`
    - :guilabel:`On save`
    - :guilabel:`On deletion`
    - :guilabel:`On UI change`

   :guilabel:`External`
    - :guilabel:`On webhook`

>    Based on the selection for the :guilabel:`Trigger`, additional fields will populate.

Reviewer 2: UPDATE: I left a comment above about this -- great to see you already had it implement. But, do you think this might be even more helpful to mention before listing off the potential 'Trigger' selections?

>    Under the :guilabel:`Before Update Domain`, set a condition to be met before updating the record.

Reviewer 2: ```suggestion
   Under the :guilabel:`Before Update Domain` field, set a condition to be met before updating the record.
```

   Click :guilabel:`Edit Domain` to set record parameters.

>    Under the :guilabel:`Actions To Do` tab, select :guilabel:`Add an action`. Next, select

Reviewer 2: ```suggestion
   Under the :guilabel:`Actions To Do` tab, select :guilabel:`Add an action`. Next, in the resulting :guilabel:`Create Actions` pop-up window, select
```

>    :guilabel:`Send SMS`, and set the :guilabel:`SMS Template`. Choose whether the SMS message should

Reviewer 2: ```suggestion
   :guilabel:`Send SMS`. Then, set the :guilabel:`SMS Template`. Choose whether the SMS message should
```
is there a reason the 'Technical Settings' section was bypassed during this explanation? Do you want to consider adding it in during your revision? Just a quick explanation should work...

>    be logged as a note, by making a selection in the dropdown menu: :guilabel:`Send SMS as`.

Reviewer 2: ```suggestion
   be logged as a note, by making a selection in the drop-down menu: :guilabel:`Send SMS as`.
```

   Click :guilabel:`Save and Close` to save the changes to this new action.

   .. image:: sms_essentials/automated-action-sms.png
      :align: center
      :alt: Automated rule template with rule to do, SMS template and log as note highlighted.

   Add any necessary notes under the :guilabel:`Notes` tab. Finally, navigate away from the

>    completed automated rule or manually save to implement the change.

Reviewer 2: ```suggestion
   completed automated rule, or manually save, to implement the change.
```
